### PR TITLE
Drop CPython 3.6 support

### PR DESCRIPTION
**Pull request by yunline**
Issue: #1850

### CODE REVIEW

**Comment by Starbuck5** [comment]
@ankith26 What do you want to do about the ubuntu-18.04 build here?

**Comment by ankith26** [comment]
If we are dropping 3.6 we also have to drop the 18.04 build (gh actions have deprecated it anyways)

**Review by MyreMylar — APPROVED** [review]
LGTM 👍 

I can't find any more CPython 3.6 references outside of a single comment (which is still relevant).

**Comment by MyreMylar** [comment]
Looking at > 3.7 stuff we could get rid of the blocks like this:

```
#if PY_VERSION_HEX < 0x03070000
#ifdef WITH_THREAD
    PyEval_InitThreads();
#endif /* WITH_THREAD */
#endif
```

**Comment by itzpr3d4t0r** [comment]
yeah the `_pygame.h` one should be related to the `FASTCALL` macros we introduced a while back for back compat with 3.6.

**Review by MyreMylar — APPROVED** [review]
Changes look good 👍

**Review by oddbookworm — APPROVED** [review]
LGTM! But seeing as how major a change this is, I'm not going to merge it because I think we might want at least one more approval?

**Review by Starbuck5 — APPROVED** [review]

File: src_c/_pygame.h [hunk 1]
```diff
@@ -281,41 +281,13 @@ supported Python version. #endif */
...
-
-#else /* PY_VERSION_HEX >= 0x03070000 */
 /* compat macros are no-op on python versions that support fastcall */
 #define PG_FASTCALL_NAME(func) func
 #define PG_DECLARE_FASTCALL_FUNC(func, self_type)
 #define PG_WRAP_FASTCALL_FUNC(func, self_type)
 
 #define PG_FASTCALL METH_FASTCALL
 
-#endif /* PY_VERSION_HEX >= 0x03070000 */
```
Comment (ankith26) [inline src_c/_pygame.h:317]: not sure whether it should be a part of this PR, or block this PR. But with this change many of these macro references can be removed in the codebase

File: .github/workflows/build-macos.yml [hunk 2]
```diff
@@ -83,10 +83,10 @@ jobs:
         # of these builds take roughly the same time
         include:
           - { 
-            name: "x86_64 (CPython 3.6 3.10 and above)",
+            name: "x86_64 (CPython 3.10 and above)",
             macarch: x86_64,
-            # pattern matches 6 or any 2 digit number
-            pyversions: "cp3{6,[1-9][0-9]}-*"
+            # pattern matches any 2 digit number
+            pyversions: "cp3{10,[1-9][0-9]}-*"
```
Comment (ankith26) [inline .github/workflows/build-macos.yml:89]:
```suggestion
            pyversions: "cp3[1-9][0-9]-*"
```
The explicit 10 here is not needed, the pattern covers for this

**Review by ankith26 — APPROVED** [review]
Overall the PR looks good, left two minor reviews for your consideration. You may choose to fix these in this PR, or (you or anyone else can) do a different PR shortly after merging this.

Thanks for the PR! :tada:

File: src_c/_pygame.h (hunk 1 above)
Comment (ankith26) [reply src_c/_pygame.h:317]: Cool!

**Pull request merged**